plugins/version-control-systems/git/src/main/kotlin/Git.kt

```diff
@@ -87,7 +86,17 @@ object GitCommand : CommandLineTool {
     override fun displayName(): String = "Git"
 }
 
-class Git internal constructor() : VersionControlSystem(GitCommand) {
+/**
+ * Provides functionality for interacting with Git repositories,
+ * utilizing either [JGit][org.eclipse.jgit.api.Git] or the Git CLI for executing operations.
+ * The following [configuration options][VersionControlSystemConfiguration.options] are available:
+ * - *submoduleHistoryDepth*: Depth of the commit history to fetch when updating submodules. Defaults to 50.
```

Comment:
This option does not only affect the history depth of submodules, but also the history depth of the top-level repository. So it should be called `historyDepth`.